catkin_ws/src/kpr_arm_control/src/arm_control/arm_control.cpp

```diff
@@ -90,33 +110,140 @@ bool sendIO(int8_t pin, float state) {
+			ROS_DEBUG("The command has been successfully perforemd, according to the input");
+			success = true;
+			break;
+		}
+		else {
+			ROS_DEBUG("The command has not been successfully perforemd, according to the input");
+		}
+	}
+	return success;
+}
+
+
+
+
+/**
  * Sends the necessary IO message to close the gripper.
  * @return True if the IO call succeeded, false otherwise.
  */
 bool startGrip() {
-	bool success = sendIO(GRIPPER_PIN, GRIPPER_CLOSE);
+	if (!sendIO(GRIPPER_PIN, GRIPPER_CLOSE)) {
+		return false;
+	}
+	ROS_DEBUG("Open command has been send to the gripper");
+	
 	//TODO Wait for IO to confirm the gripper has closed.
 	// Attempt to close multiple times in slightly different conditions
 	// if it failed.
-	return success;
+	return true;
 }
 
 /**
  * Sends the necessary IO message to do a cut.
- * @return True if the IO call succeeded, false otherwise.
+ * @return True if the IO calls succeeded and the sensors confirm the expected result, false otherwise.
  */
 bool cut() {
 	if (!sendIO(CUTTER_PIN, CUTTER_CLOSE)) {
 		return false;
 	}
-	//TODO Add wait for IO to confirm the cutter has closed.
-	return sendIO(CUTTER_PIN, CUTTER_OPEN);
+	ROS_DEBUG("Close command has been send to the cutter");
+	
+	bool success = false;
+	for (int tries = 0; tries <= 10; tries++) {
+		success = readDigInput(CUT_CLOSED_STATE, pinstates);
+		if (success) {
+			break; 
+		}
+			ROS_DEBUG("Cutter is not open according to IO and a new attempt would need to be made");
+			//TODO Add new commands due to failed attempt.
+	}
+	if (success == false) {
+		return false;
+	}
+		
+	if (!sendIO(CUTTER_PIN, CUTTER_OPEN)) {
+		return false;
+	}
+	ROS_DEBUG("Open command has been send to the cutter");
+
+	for (int tries = 0; tries <= 10; tries++) {
+		success = readDigInput(CUT_OPEN_STATE, pinstates);
+		if (success) {
+			break; 
+		}
+			ROS_DEBUG("Cutter is not open according to IO and a new attempt would need to be made");
+			//TODO Add new command due to failed attempt.
+	}
+	if (success == false) {
+		return false;
+	}
 }
 
 /**
  * Sends the necessary IO message to open the gripper.
  * @return True if the IO call succeeded, false otherwise.
  */
 bool releaseGrip() {
-	return sendIO(GRIPPER_PIN, GRIPPER_OPEN);
+	if (!sendIO(GRIPPER_PIN, GRIPPER_OPEN)) {
+		return false;
+	}
+	//TODO Wait for IO to confirm the gripper has closed.
+}
+
+/**
+ * Sends the necessary IO message to enable the vacuum suction.
+ * @return True if the IO call succeeded and the sensor confirms the expected result, false otherwise.
+ */
+bool startVacuum() {
+	if (!sendIO(VACUUM_PIN, VACUUM_ON)) {
+		return false;
+	}
+	
+	bool success = false;
+	for (int tries = 0; tries <= 10; tries++) {
+		success = readDigInput(PRESSURE_STATE, pinstates);
+		if (success) {
+			return true;
+			break; 
+		}
+			ROS_DEBUG("Suction is not correct according to IO and a new attempt would need to be made");
+			//TODO Add new commands due to failed attempt.
+	}
+	if (success == false) {
```

Comment:
`success` seems to be redundant in this function. Lines 221 and 222 could be `if (readDigInput(PRESSURE_STATE, pinstates) {` and 229 to 231 just `return false`.